components/dashboard/src/settings/SSHKeys.tsx

```diff
@@ -0,0 +1,257 @@
+        } catch (e) {
+            setErrorMsg(e.message);
+            return false;
+        }
+        props.onClose();
+        props.onSave();
+        return true;
+    };
+
+    return (
+        <Modal
+            title="New SSH Key"
+            buttons={
+                <button className="ml-2" onClick={save}>
+                    Add Key
+                </button>
+            }
+            visible={true}
+            onClose={props.onClose}
+            onEnter={save}
+        >
+            <>
+                {errorMsg.length > 0 && (
+                    <Alert type="error" className="mb-2">
+                        {errorMsg}
+                    </Alert>
+                )}
+            </>
+            <div className="text-gray-500 dark:text-gray-400 text-md">
+                Add an SSH key for secure access workspaces via SSH. Learn more
+            </div>
+            <Alert type="info" className="mt-2">
+                SSH key are used to connect securely to workspaces.{" "}
+                <a
+                    href="https://www.gitpod.io/docs/configure/ssh#create-an-ssh-key"
+                    target="gitpod-create-ssh-key-doc"
+                    className="gp-link"
+                >
+                    Learn how to create an SSH Key
+                </a>
+            </Alert>
+            <div className="mt-2">
+                <h4>Key</h4>
+                <textarea
+                    autoFocus
+                    style={{ height: "160px" }}
+                    className="w-full resize-none"
+                    value={value.key}
+                    placeholder="Begins with 'ssh-rsa', 'ecdsa-sha2-nistp256',
+                        'ecdsa-sha2-nistp384', 'ecdsa-sha2-nistp521',
+                        'ssh-ed25519',
+                        'sk-ecdsa-sha2-nistp256@openssh.com', or
+                        'sk-ssh-ed25519@openssh.com'"
+                    onChange={(v) => update({ key: v.target.value })}
+                />
+            </div>
+            <div className="mt-2">
+                <h4>Title</h4>
+                <input
+                    className="w-full"
+                    type="text"
+                    placeholder="e.g. laptop"
+                    value={value.name}
+                    onChange={(v) => {
+                        update({ name: v.target.value });
+                    }}
+                />
+            </div>
+        </Modal>
+    );
+}
+
+export function DeleteSSHKeyModal(props: DeleteModalProps) {
+    const confirmDelete = async () => {
+        await getGitpodService().server.deleteSSHPublicKey(props.value.id!);
+        props.onConfirm();
+        props.onClose();
+    };
+    return (
+        <ConfirmationModal
+            title="Delete SSH Key"
+            areYouSureText="This action CANNOT be undone. This will permanently delete the SSH key and if you'd like to use it in the future, you will need to upload it again."
```

Comment:
**suggestion**: We could go with a less scary warning here, unless you think this is really needed here. Updating the copy later is also an option. Also, including the key title and last used timestamp could help associate the key about to be deleted. What do you think?

```suggestion
            areYouSureText="Are you sure you want to delete this SSH Key?"
```

| Delete SSH Key Modal |
|-|
| <img width="1440" alt="Screenshot 2022-06-16 at 7 02 00 PM" src="https://user-images.githubusercontent.com/120486/174115222-07ced570-b8c4-457c-951a-cc471eb16a81.png"> |

Reply:
Good for me, the previous error message is a copy from GH

Reply:
> ... the previous error message is a copy from GH

Yes, I noticed that. 🙂 

I'd think it's ok to go with our own copy style found in the product for now and adjust later if needed.

Thank you, @mustard-mh! 🙏